src/components/v5/common/Fields/Select/Select.tsx

```diff
@@ -83,9 +82,9 @@ const Select = <T extends any[]>({
     list.filter(({ value }) => value !== selectedItem.value);
   const optionsList = isPlaceholder ? list : filteredList;
 
-  const placeholder = getFormattedText(placeholderText);
-  const selectedLabel = selectedItem
-    ? getFormattedText(selectedItem.label)
+  const placeholder = placeholderText ? formatText(placeholderText) : undefined;
```

Comment:
Why doesn't `formatText` handle this case internally as well? 

You feed it some data, and if it's not what it expects, it just returns undefined. That way, you won't have these checks all over the place.